pkg/cpio/newc/newc.go

```diff
@@ -117,6 +117,9 @@ func (w *writer) WriteRecord(f cpio.Record) error {
 
 	buf := &bytes.Buffer{}
 	hdr := headerFromInfo(f.Info)
+	if f.ReadCloser == nil {
```

Comment:
Same thing here: this is an issue with how the records are created in the first place. Go sets struct fields to their null value by default, so FileSize is always 0 by default. You have to explicitly set it to something non-zero. So if we get a nil ReadCloser here, but a FileSize != 0, that means someone is screwing up a Record passed in here. We should fix where that Record is created, not add another conditional here.